File: Demo/ObjcDemo/CustomPageController.m
//
//  CustomPageController.m
//  ObjcDemo
//
//  Created by kelin on 2020/11/10.
//

#import "CustomPageController.h"
#import "KKAutoCycleScrollView.h"
#import <SDWebImage/SDWebImage.h>
@interface CustomPageController ()<KKAutoCycleScrollViewDelegate, KKAutoCycleScrollViewDataSource>
@property (nonatomic, strong) KKAutoCycleScrollView *cycleView;
@end

@implementation CustomPageController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    self.title = @"自定义每个page";
    [self initCycleView];
}

- (void)initCycleView {
    _cycleView = [[KKAutoCycleScrollView alloc] initWithFrame:CGRectMake(0, 100, self.view.frame.size.width, 200)];
    _cycleView.delegate = self;
    _cycleView.dataSource = self;
    _cycleView.duration = 4;
    [_cycleView create];
    
    [self.view addSubview:_cycleView];
}

- (void)cycleView:(KKAutoCycleScrollView *)cycleView addSubViewsInPageView:(UIView *)pageView subViewAtIndex:(NSInteger)index {

    if (index == 0) {
        UIImageView *imageView = [[UIImageView alloc] initWithFrame:pageView.bounds];
        imageView.image = [UIImage imageNamed:@"1.png"];
        [pageView addSubview:imageView];
    }
    
    if (index == 1) {
        
        pageView.backgroundColor = [UIColor colorWithRed:240/255.0 green:240/255.0 blue:240/255.0 alpha:1.];
        
        UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake((pageView.frame.size.width - 200)/2, 30, 200, 60)];
        label.backgroundColor = [UIColor blackColor];
        label.textColor = [UIColor whiteColor];
        label.font = [UIFont boldSystemFontOfSize:16];
        label.textAlignment = NSTextAlignmentCenter;
        label.text = @"这这这..是一个label";
        [pageView addSubview:label];
        
        for (NSInteger i = 0; i < 3; i++) {
            CGFloat random1 = (arc4random() % 255) / 255.0;
            CGFloat random2 = (arc4random() % 255) / 255.0;
            CGFloat random3 = (arc4random() % 255) / 255.0;
            UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
            button.frame = CGRectMake(20 * (i + 1) + 80 * i, CGRectGetMaxY(label.frame) + 10, 80, 40);
            button.backgroundColor = [UIColor colorWithRed:random1 green:random2 blue:random3 alpha:1.];
            [button setTitle:[NSString stringWithFormat:@"button%ld",i+1] forState:UIControlStateNormal];
            [pageView addSubview:button];
        }
        
    }
    
    if (index == 2) {
        UIImageView *imageView = [[UIImageView alloc] initWithFrame:pageView.bounds];
        [imageView sd_setImageWithURL:[NSURL URLWithString:@"https://icweiliimg6.pstatp.com/weili/l/378712400698146835.jpg"]];
        [pageView addSubview:imageView];
    }
    
}

- (NSInteger)numberOfViewCountInCycleView:(KKAutoCycleScrollView *)cycleView {
    return 3;
}

- (void)cycleView:(KKAutoCycleScrollView *)cycleView didClickPageViewAtIndex:(NSInteger)index {
    NSLog(@"current page is %ld", index);
}

- (void)dealloc {
    [_cycleView destory];
}


@end

File: Demo/SwiftDemo/CustomPageController.swift
//
//  CustomPageController.swift
//  SwiftDemo
//
//  Created by kelin on 2020/11/10.
//

import Foundation
import UIKit
import SDWebImage

class CustomPageController: UIViewController, KKAutoCycleScrollViewDelegate, KKAutoCycleScrollViewDataSource {
    
    var cycleView:KKAutoCycleScrollView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "自定义每个page"
        self.view.backgroundColor = UIColor.white
        
        self.initCycleView()
    }
    
    func initCycleView() -> Void {

        self.cycleView = KKAutoCycleScrollView(frame: CGRect(x: 0, y: 100, width: self.view.frame.size.width, height: 200))
        self.cycleView.delegate = self;
        self.cycleView.dataSource = self;
        self.cycleView.duration = 2
        self.cycleView.create()
        
        self.view.addSubview(self.cycleView)
    }
    
    func numberOfViewCountInCycleView(cycle: KKAutoCycleScrollView) -> NSInteger {
        return 3
    }
    
    func cycleViewAddSubViewsInPageViewAtIndex(cycle: KKAutoCycleScrollView, pageView: UIView, index: NSInteger) {
        
        if index == 0 {
            let imageView = UIImageView(frame: pageView.bounds)
            imageView.image = UIImage(named: "1.png")
            pageView.addSubview(imageView)
        }
        
        if index == 1 {
            pageView.backgroundColor = UIColor.init(red: 240/255.0, green: 240/255.0, blue: 240/255.0, alpha: 1.0)
            let label = UILabel(frame: CGRect(x: (pageView.frame.size.width - 200)/2, y: 30, width: 200, height: 60))
            label.backgroundColor = UIColor.black
            label.textColor = UIColor.white
            label.font = UIFont.boldSystemFont(ofSize: 16)
            label.textAlignment = .center
            label.text = "这这这..是一个label"
            pageView.addSubview(label)
            
            for i in 0...2 {
                let random1:CGFloat = CGFloat((arc4random() % 255)) / 255.0
                let random2:CGFloat = CGFloat((arc4random() % 255)) / 255.0
                let random3:CGFloat = CGFloat((arc4random() % 255)) / 255.0
                let button = UIButton(type: .system)
                button.frame = CGRect(x: CGFloat(20 * (i + 1) + (80 * i)), y: label.frame.maxY + 10, width: 80, height: 40)
        
                button.backgroundColor = UIColor.init(red: random1, green: random2, blue: random3, alpha: 1.0)
                button.setTitle("button\(i+1)", for: .normal)
                pageView.addSubview(button)
            }
        }
        
        if index == 2 {
            let imageView = UIImageView(frame: pageView.bounds)
            imageView.sd_setImage(with: URL(string: "https://icweiliimg6.pstatp.com/weili/l/378712400698146835.jpg"), completed: nil)
            pageView.addSubview(imageView)
        }
        
    }
    
    func cycleViewDidClickPageViewAtIndex(cycleView: KKAutoCycleScrollView, index: NSInteger) {
        print("current page is %ld", index)
    }
    
    deinit {
        self.cycleView.destory()
    }
}

